lib/components/base/ProjectCard.vue

```diff
@@ -83,147 +86,69 @@ import {
 import dayjs from 'dayjs'
 import relativeTime from 'dayjs/plugin/relativeTime.js'
 dayjs.extend(relativeTime)
-</script>
 
-<script>
-import { defineComponent } from 'vue'
-export default defineComponent({
-  props: {
-    id: {
-      type: String,
-      default: 'modrinth-0',
-    },
-    type: {
-      type: String,
-      default: 'mod',
-    },
-    name: {
-      type: String,
-      default: 'Project Name',
-    },
-    author: {
-      type: String,
-      default: null,
-    },
-    description: {
-      type: String,
-      default: 'A _type description',
-    },
-    iconUrl: {
-      type: String,
-      default: '#',
-      required: false,
-    },
-    downloads: {
-      type: String,
-      default: null,
-      required: false,
-    },
-    follows: {
-      type: String,
-      default: null,
-      required: false,
-    },
-    createdAt: {
-      type: String,
-      default: '0000-00-00',
-    },
-    updatedAt: {
-      type: String,
-      default: null,
-    },
-    categories: {
-      type: Array,
-      default() {
-        return []
-      },
-    },
-    filteredCategories: {
-      type: Array,
-      default() {
-        return []
-      },
-    },
-    projectTypeDisplay: {
-      type: String,
-      default: null,
-    },
-    projectTypeUrl: {
-      type: String,
-      default: null,
-    },
-    status: {
-      type: String,
-      default: null,
+/** Creates a type that contains original T string, but allows other string values as well. */
+type LaxString<T extends string> = T | (string & Record<never, never>)
+
+type ProjectType = 'mod' | 'plugin' | 'modpack' | 'resourcepack'
+type SideRequirement = 'required' | 'optional' | 'unsupported'
+
+const props = withDefaults(
+  defineProps<{
+    id: string
+    type: LaxString<ProjectType>
+    name: string
```

Comment:
shouldn't these (id, type, name) also have a "?"´

since they aren't required because they have default values

Reply:
Good eye, yeah. Will fix that.